Assets/Scripts/Game/PlayerMouseLook.cs

```diff
@@ -86,6 +97,76 @@ public float PitchMinLimit
             set { pitchMin = Mathf.Clamp(value, PitchMin, PitchMax); Pitch = Pitch; }
         }
 
+        // Scales fractional progression (non-linear) to frame rate
+        private float GetFrameRateScaledFractionOfProgression(float fractionAt60FPS)
+        {
+            float frames = Time.unscaledDeltaTime * 60f; // Number of frames to handle this tick, can be partial
+            float c = (1.0f - fractionAt60FPS) / fractionAt60FPS;
+            return 1.0f - c / (frames + c);
+        }
+
+        // Applies scaled raw mouse deltas to lookTarget, then calls ApplySmoothing method to update lookCurrent
+        void ApplyLook()
+        {
+            // Scale sensitivity
+            float sensitivityX = 1.0f;
+            float sensitivityY = 1.0f;
+
+            if (InputManager.Instance.UsingController)
+            {
+                // Make sure it keeps consistent speed regardless of framerate
+                // Speed = speed * 60 frames / (1 / unscaledDeltaTime) or speed * 60 * unscaledDeltaTime
```

Comment:
I was wondering what happens if user has a high refresh monitor (e.g. 120fps or higher)? Does the new smoothing method still scale to very high refresh rates?

Reply:
Yes. But you don't have to take my word for it!

I've added a block of test code so you can try it out yourself (if you wish).

Pressing LEFT CTRL key activates the test.

Here's what the test code does:
1. Sets smoothing to a high value (0.99f) to give an exaggerated, more drawn-out effect
2. Levels pitch, then sets lookCurrent yaw to 0° and lookTarget yaw to 90°
3. Monitors lookCurrent and finishes test when lookCurrent yaw reaches 60° (two-thirds of the way to lookTarget)
4. Displays a popup showing how much time elapsed and what the average FPS were over that period

Once you've run the test you've got a reference point.

There are three ways of getting additional reference points for comparison:

1. Change your display's physical refresh rate or FPS limit and repeat the test (time should be the same)
2. Uncomment the "hang time" code at the bottom of the test function to lower FPS (time should be the same)
3. *Simulate* a different frame rate by adding a multiplier to Time.unscaledDeltaTime in the PlayerMouseLook.GetFrameRateScaledFractionOfProgression method

For the third option:

a) If you doubled delta time (multiplied by 2) then the test should take half as long to complete (simulating half frame rate so do twice as much per update)
b) If you halved delta time (multiplied by 0.5) then the test should take twice as long to complete (because you're simulating 120 FPS but are actually only running at 60 FPS)

So:

`float frames = Time.unscaledDeltaTime * 60f;`

Becomes something like:

`float frames = Time.unscaledDeltaTime * 2 * 60f;`

If the times come out as expected then the scaling is correct.

Results using option #2:

![test_time_at_55_fps](https://user-images.githubusercontent.com/115361770/195961593-4a3a9e53-dd41-44c5-8c45-ba1fe60ef096.png)
![test_time_at_47_fps](https://user-images.githubusercontent.com/115361770/195961622-b9128648-f674-41a6-8351-3b311f6e2a2a.png)
![test_time_at_34_fps](https://user-images.githubusercontent.com/115361770/195961635-6b19d2b7-4165-45ba-98ed-49eb501b0dc6.png)
![test_time_at_22_fps](https://user-images.githubusercontent.com/115361770/195961654-17e08470-81a6-460e-897d-b9d10adf591b.png)

You asked specifically about higher FPS (120+) and the results check out using option #3.

Reply:
Simulating 120 FPS at 60 FPS (time should be doubled):

![test_time_simulated_120_fps](https://user-images.githubusercontent.com/115361770/195993634-ef89a1f4-3e7f-439a-b04f-0cc18edd6b40.png)

Simulating 300 FPS at 60 FPS (time should be five times longer):

![test_time_simulated_300_fps](https://user-images.githubusercontent.com/115361770/195993678-d54c0a83-d4e9-4ee7-8c87-66bbc25a0291.png)

That last result is slightly higher than the expected 9.25 sec but still quite acceptable.